Explicitly set the txn priority in TestPropagateTxnOnPushError

kkaneda opened the pull request:
I couldn't reproduce a failure locally by "make stress PKG=./kv", so this will probably not fix a stress test failure we saw (#4906), but I wanted to give a shot.

Also fix comments, clean up the test, and add some logging. The test was updated with 2a858b06, but most of the comments were not updated.

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Review on Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/5013)

<!-- Reviewable:end -->

kkaneda committed:
Explicitly set the txn priority in TestPropagateTxnOnPushError

I couldn't reproduce a failure locally by "make stress PKG=./kv", so
this will probably not fix a stress test we saw (#4906), but I wanted
to give a shot.

Also fix comments, clean up the test, and add some logging. The test
was updated with 2a858b06, but most of the comments were not updated.

tschottdorf commented on kv/dist_sender_server_test.go:
```diff
@@ -550,16 +550,23 @@ func TestPropagateTxnOnPushError(t *testing.T) {
 		if pErr := db.Txn(func(txn *client.Txn) *roachpb.Error {
-			if pErr := txn.Put("b", "val"); pErr != nil {
+			// Set high priority so that the intent will not be pushed.
+			txn.InternalSetPriority(highriority)
```
typo

kkaneda replied:
fixed..

tschottdorf commented:
Hmm. Doesn't repro here as well - odd. Do you have any theory regarding the failure?

BTW, if you write "does not fix ..." then all Github will see is "fix ..." and so the issue would be closed on merge.

kkaneda commented:
Sure, I don't have a very good explanation, but the test failed since the second txn wasn't retried. Possible reasons for that are:

- the goroutine didn't write an intent.
- the second transaction pushed the intent written by the txn created from the goroutine.

For the first case, I've added more log messages and changed the test to always wait for the completion  of the goroutine, so that we will have more info when the test fails again.

For the second case, it could happen when the second txn has a higher priority than the first txn. So, I explicitly set the priority to both txns. There might be other reason why the push succeeded, though.

> BTW, if you write "does not fix ..." then all Github will see is "fix ..." and so the issue would be closed on merge.

Ah, thanks! Tweaked the message.

tschottdorf commented:
LGTM. I'm fine with closing the issue. If it comes back, we'll see.

kkaneda commented:
Thanks! Merging now. Will reopen the issue if the test fails again.
